qos/evm/check_blocknumber.go

```diff
@@ -0,0 +1,56 @@
+package evm
+
+import (
+	"fmt"
+
+	"github.com/buildwithgrove/path/qos/jsonrpc"
+)
+
+// methodBlockNumber is the JSON-RPC method for getting the latest block number.
+// Reference: https://ethereum.org/en/developers/docs/apis/json-rpc/#eth_blocknumber
+const methodBlockNumber = jsonrpc.Method("eth_blockNumber")
+
+var (
+	errNoBlockNumberObs      = fmt.Errorf("endpoint has not had an observation of its response to a %q request", methodBlockNumber)
+	errInvalidBlockNumberObs = fmt.Errorf("endpoint returned an invalid response to a %q request", methodBlockNumber)
+)
+
+// endpointCheckBlockNumber is a check that ensures the endpoint's block height within the sync allowance.
+// It is used to ensure that the endpoint is not behind the chain.
+//
+// Note that this check does not have an expiry as it is expected to be run frequently.
+// This serves two purposes:
+//   - It ensures that the endpoint is not behind the chain.
+//   - It ensures that the hydrator is always sending at least some request to enforce protocol-level sanctions.
+type endpointCheckBlockNumber struct {
+	// parsedBlockNumberResponse stores the result of processing the endpoint's response to an `eth_blockNumber` request.
+	// It is nil if there has NOT been an observation of the endpoint's response to an `eth_blockNumber` request.
+	parsedBlockNumberResponse *uint64
+}
+
+// isValid returns an error if the endpoint's block height is less than the perceived block height minus the sync allowance.
+func (e *endpointCheckBlockNumber) isValid(perceivedBlockNumber uint64, syncAllowance uint64) error {
```

Comment:
Ditto re placement of logic: it is the responsibility of serviceState.
To further expand:
On `main` branch, we have: check -> request -> observation -> serviceState update 
With this PR, the checks are getting overloaded.

Reply:
Makes sense, I've moved the logic to the service state.